File: Solution/Python/849. Maximize Distance to Closest Person.py
class Solution:
    def maxDistToClosest(self, seats):
        """
        :type seats: List[int]
        :rtype: int
        """
        d = {}
        res = l = left = r = right = 0
        for i, s in enumerate(seats):
            if not s and left:
                d[i] = l = l + 1
            elif s:
                l, left = 0, 1
        for i in range(len(seats) - 1, -1, -1):
            if not seats[i] and right and (i not in d or d[i] > r):
                d[i] = r = r + 1
            elif seats[i]:
                r, right = 0, 1
        return max(d.values())


seats = [1, 0, 0, 0, 1, 0, 1]
res = Solution().maxDistToClosest(seats)
print(res)

File: Solution/Javascript/849. Maximize Distance to Closest Person.js
/**
* @param {number[]} seats
* @return {number}
*/
var maxDistToClosest = function (seats) {
    let continueArr = [];
    let continueCount = 0;
    for (let i = 0; i < seats.length; i++) {
        if (seats[i] == 0) {
            continueCount++;
        }
        if ((seats[i] == 1 && continueCount > 0) || i == seats.length - 1) {
            continueArr.push(continueCount);
            continueCount = 0;
        }
    }

    let arr = continueArr.map((val) => Math.ceil(val / 2))

    let leftContinue = 0
    if (seats[0] == 0) {
        leftContinue = continueArr[0];
    }
    let rightContinue = 0
    if (seats[seats.length - 1] == 0) {
        rightContinue = continueArr[continueArr.length - 1];
    }

    return Math.max(leftContinue, rightContinue, Math.max.apply(null, arr));
}

// let seats = [1, 0, 0, 0, 1, 0, 1];
// let seats = [1, 0, 0, 0, 0, 1, 0, 1, 0, 0, 0];
// let seats = [0, 1]
let seats = [0, 0, 1]
// let seats = [1, 0, 0, 0]
// let seats = [0, 1, 0, 1, 0]
let res = maxDistToClosest(seats);
console.log(res);
